hotel_booking.py

```diff
@@ -0,0 +1,210 @@
+                                     'customer_type','reservation_status']]
+df_cat.info()
+
+# Creating a new dataframe for numerical data
+df_num= df2.drop(['hotel','is_canceled', 'arrival_date_month','meal',
+                                       'country','market_segment','distribution_channel',
+                                       'is_repeated_guest', 'reserved_room_type',
+                                       'assigned_room_type','deposit_type','agent',
+                                       'customer_type','reservation_status'], axis = 1)
+df_num.info()
+
+# Checking the correlation between Categorical Data by heatmap
+cor_matrix_cat = df_cat.corr()
+plt.figure(figsize=(10,10))
+cat_cor_mask = np.triu(np.ones_like(cor_matrix_cat))
+sns.heatmap(cor_matrix_cat, annot=True, fmt=".2f", cmap='RdBu', mask= cat_cor_mask, vmin=-1, vmax=1, center= 0, square=True, linewidths=2, cbar_kws={"shrink": .5}).set(ylim=(17, 0))
+plt.title("Correlation Matrix for Categorical Data ",size=15, weight='bold')
+# the reservation_status has a high reverse correlation with is_canceled and it seemed we have to delete that column
+
+# Checking the correlation between Numerical heatmap
+plt.figure(figsize=(12,12))
+cor_matrix_num = df_num.corr()
+mask_numerical = np.triu(np.ones_like(cor_matrix_num, dtype=np.bool))
+sns.heatmap(cor_matrix_num, annot=True, fmt=".2f", cmap='RdBu', mask= mask_numerical, vmin=-1, vmax=1, center= 0, square=True, linewidths=2, cbar_kws={"shrink": .5}).set(ylim=(17, 0))
+plt.title("Correlation Matrix for Numerical Data ",size=15, weight='bold')
+# babies has not an important correlation.
+
+df3 = df2.drop(['reservation_status', 'babies', 'reservation_status_date','company'], axis=1)
+
+# Building the sample group
+y_model = df2.iloc[:,1]
+X_model = pd.concat([df2.iloc[:,0],df3.iloc[:,2:]], axis=1)
+y_model.describe()
+
+# Split to train and test with 70-30 ratio
+from sklearn.model_selection import train_test_split
+X_train, X_test, y_train, y_test = train_test_split(X_model, y_model, test_size=0.3, random_state=42)
+
+# Implement standart scaler method
+from sklearn.preprocessing import StandardScaler
+standardScalerX = StandardScaler()
+X_train = standardScalerX.fit_transform(X_train)
+X_test = standardScalerX.fit_transform(X_test)
+
+# Decision Tree Model Building
+from sklearn.tree import DecisionTreeClassifier
+decision_tree_m = DecisionTreeClassifier(criterion= 'gini', min_samples_split=8, min_samples_leaf = 4, max_features = 'auto')
+# fitting the model
+decision_tree_m.fit(X_train, y_train)
+
+from sklearn.metrics import classification_report
+# Predicting Model
+predict_decision_tree = decision_tree_m.predict(X_test)
+
+# Classification Reports
+print("Decision Tree Classification: \n",classification_report(y_test, predict_decision_tree))
+print("Accuracy on test data: {:.2f}".format(decision_tree_m.score(X_test, y_test)))
+plt.show()
+print("DONE")
```

Comment:
![image](https://user-images.githubusercontent.com/94354175/176889064-cf48ba9f-45d5-4125-8bc6-709c1b460b60.png)
If you are wondering, this icon means the file does not end with a new line. If you are fixing other things, please add that too. It does not really matter, expect for when you add a new line of code under it. Then git will think that you changed this line too (because of the added new-line character).